### ISSUE TITLE
Bokeh "expand to fullscreen" for demonstration purposes.

### ISSUE THREAD

**Issue by nfaggian** [bokeh/bokeh#3598]
Do you think it would be good to have a feature that allows a bokeh plot to explode out to fill the screen? 

An example of where this sort of thing works is in the diagramming tool "lucid charts", which has a feature like this and it is *very* helpful when demonstrating work to people.

I think this could be a widget that you can click on like any of the other tools.

**Comment by bryevdv** [comment]
ping @birdsarah is this something that might fall naturally out of the `0.12` layout work?

**Comment by birdsarah** [comment]
I don't think it falls out of the layout work, but its a nice feature idea and seems like it would be an easy tool to add. 

The layout work may make it slightly easier, but i'd say it's largely irrelevant.

**Comment by nfaggian** [comment]
Great to see this is still on people's radar. I hope someone smart can pull it off :)

**Comment by bryevdv** [comment]
ref: https://stackoverflow.com/questions/1125084/how-to-make-the-window-full-screen-with-javascript-stretching-all-over-the-scre

**Comment by rabernat** [comment]
Thanks to the bokeh devs for this amazing tool! I appreciate all the time and effort you put into this project.

👍 to this issue. I would really enjoy having a fullscreen mode on my holoviews bokeh plots.

**Comment by bobcolner** [comment]
this would be a great feature.

**Comment by philippjfr** [comment]
@bryevdv @mattpap How would you feel about adding a tool based on the [screenfull.js library](https://github.com/sindresorhus/screenfull.js)? The implementation is pretty straightforward and I don't think the dependency is very big.

**Comment by bryevdv** [comment]
2k minified? 👍

**Comment by mattpap** [comment]
>  How would you feel about adding a tool based on the screenfull.js library?

I think it's unnecessary, primarily because this library doesn't do anything besides prefixing API for browser compatibility. 2k minified is a pretty steep price for this, but more importantly that's another dependency.

**Comment by bryevdv** [comment]
There are lots of ways to pay: time, for instance. So, it's necessary to the extent you are willing to pay the time to do without it. Otherwise, I am +1 on this.

**Comment by MarcSkovMadsen** [comment]
I have been hoping for something like this for a long time @philippjfr . Maybe it's not a Bokeh thing but a Panel thing?

But as a user it is so nice to be able to maximize things and focus on them. I thought it was something I would have to support for example at awesome-panel.org using a custom template.

Alternative would be to have a panel maximize layout or a maximize attribute for all panes and layouts.

Even SAP has this functionality as you can see below :-)

https://sapui5.hana.ondemand.com/#/entity/sap.viz.ui5.controls.VizFrame/sample/sap.viz.sample.VerticalBullet

![image](https://user-images.githubusercontent.com/42288570/84499484-fbfc6600-acb2-11ea-85fd-f5c9c0f8679a.png)

**Comment by Assistedevolution** [comment]
+1 interested in  this feature. Would be very useful from an end user perspective

**Comment by mattpap** [comment]
We know, there were certain design limitations that prevented this, but those are currently being worked on, so we should be able to add this feature in future.

**Comment by MarcSkovMadsen** [comment]
Checking in on this as it would be super nice to have. The Fast templates of Panel have panel with "full screen" functionality. But to really take advantage of this we would need `responsive=True` to work again for bokeh plots. It does not currently work due to the bokeh regression.


https://user-images.githubusercontent.com/42288570/121537576-57674e00-ca04-11eb-96ba-51589891e1e1.mp4

**Comment by bryevdv** [comment]
> But to really take advantage of this we would need responsive=True to work again for bokeh plots. It does not currently work due to the bokeh regression.

This is a confusing comment. The `responsive` parameter was *removed* (and is not coming back) because a two-state Boolean flag is not sufficient to encapsulate all the possibilities for layout behavior. I am not sure how it relates to this issue at all. 

We could consider reviving #9563 Given that an alternative has not materialized in 18 months I would support adding the small dependency now, even if it is intended to be replaced at some point. cc @philippjfr @mattpap

**Comment by philippjfr** [comment]
>This is a confusing comment. The responsive parameter was removed (and is not coming back) because a two-state Boolean flag is not sufficient to encapsulate all the possibilities for layout behavior. I am not sure how it relates to this issue at all.

The context is that HoloViews does support a `responsive` flag, which sets the sizing_mode depending on whether width/height are set. Anyway, I don't see how this relates to this issue either, but I've reminded @mattpap about the layout regression that this comment was referring to.

>We could consider reviving #9563 Given that an alternative has not materialized in 18 months I would support adding the small dependency now, even if it is intended to be replaced at some point. cc @philippjfr @mattpap

I'll leave that up to @mattpap since he indicated it would soon be easier to implement this with a dependency.

**Comment by mattpap** [comment]
A fullscreen tool was added in PR #11915, but the implementation requires some tweaking to make it actually useful.

**Comment by AlbertoFabbri93** [comment]
Has fullscreen mode been implemented? I don't see such a button when I use bokeh in my code 🤔

**Comment by mattpap** [comment]
It was, but only reliably for single plots. You have to add `FullscreenTool` (`fullscreen` alias) to your plots to have access to it.
